src/fastener.js

```diff
@@ -126,9 +126,17 @@ function downToU(key, z) {
   if (isArray(focus) && isNumber(key) && 0 <= key && key < focus.length)
     return fromArray(focus, key, dissocPartialU("focus", z))
 }
-
 export const downTo = curry2(downToU)
 
+function downPathU([k, ...ks], z) {
+  const { focus } = z
+  if ((isObject(focus) || isArray(focus)) && k in focus) {
+    const d = downToU(k, z)
+    return ks.length > 0 ? downPathU(ks, d) : d
+  }
+}
```

Comment:
I replaced it w/ your implementation.

As for what to do w/ `undefined` zipper args I don't have a strong opinion. Sounds like you want to treat `undefined` as a `Nothing` value.